src/frameworks/ccpa_from_us_privacy_string.js

```diff
@@ -0,0 +1,47 @@
+const FrameworkBase = require('./base');
+
+/**
+ * Implements usprivacy string framework
+ * for more information on the US Privacy string see:
+ * https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/US%20Privacy%20String.md#us-privacy-string-format
+ */
+class CCPAFromUSPrivacyString extends FrameworkBase {
+  constructor() {
+    super();
+    this.usPrivacyString = '';
+  }
+
+  useConfig({ usp }) {
+    if (usp) {
+      this.usPrivacyString = ('' + usp).toUpperCase();
+    }
+  }
+
+  supportedCapabilities() {
+    return ['canUsePersonalInformationForTargeting', 'hasBeenNotifiedOfRights'];
+  }
+
+  canUsePersonalInformationForTargeting() {
+    return this.consentStringAllowsPersonalDataSale();
+  }
+
+  hasBeenNotifiedOfRights() {
+    return this.consentStringAcknowledgesUserHasBeenNotifiedOfRights();
+  }
+
+  consentStringAllowsPersonalDataSale() {
+    if (!this.supportedUsPrivacyStringVersion()) return true;
+    if (!this.consentStringAcknowledgesUserHasBeenNotifiedOfRights()) return true;
+    return this.usPrivacyString[2] !== 'Y';
+  }
+
+  consentStringAcknowledgesUserHasBeenNotifiedOfRights() {
+    return this.supportedUsPrivacyStringVersion() && this.usPrivacyString[1] === 'Y';
+  }
+
+  supportedUsPrivacyStringVersion() {
+    return this.usPrivacyString.length === 4 && this.usPrivacyString[0] === '1';
```

Comment:
Should this check for presence of `this.usPrivacyString` first? Or is it sufficient enough to assume that `useConfig` will have been called before this?

Reply:
Oh, this is a good point, something I addressed in the branch [`with-generators`](https://github.com/voxmedia/data-privacy-compliance/blob/with-generators/src/frameworks/ccpa_from_us_privacy_string.js#L22) and need to port into this branch. Thank you.

Reply:
okay I have committed https://github.com/voxmedia/data-privacy-compliance/pull/1/commits/798f65771a37a6eb77b186c493c6164863285a1c which addresses this issue, thank you Drew